tests/storage/migration.go

```diff
@@ -0,0 +1,320 @@
+		}
+		waitMigrationToNotExist := func(vmiName, ns string) {
+			Eventually(func() bool {
+				ls := labels.Set{
+					virtv1.VolumesUpdateMigration: vmiName,
+				}
+				migList, err := virtClient.VirtualMachineInstanceMigration(ns).List(context.Background(),
+					metav1.ListOptions{
+						LabelSelector: ls.String(),
+					})
+				Expect(err).ToNot(HaveOccurred())
+				if len(migList.Items) == 0 {
+					return true
+				}
+				return false
+
+			}, 120*time.Second, time.Second).Should(BeTrue())
+		}
+		waitVMIToHaveVolumeChangeCond := func(vmiName, ns string) {
+			Eventually(func() bool {
+				vmi, err := virtClient.VirtualMachineInstance(ns).Get(context.Background(), vmiName,
+					metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				conditionManager := controller.NewVirtualMachineInstanceConditionManager()
+				return conditionManager.HasCondition(vmi, virtv1.VirtualMachineInstanceVolumesChange)
+			}, 120*time.Second, time.Second).Should(BeTrue())
+		}
+
+		waitForMigrationToSucceed := func(vmiName, ns string) {
+			waitMigrationToExist(vmiName, ns)
+			Eventually(func() bool {
+				vmi, err := virtClient.VirtualMachineInstance(ns).Get(context.Background(), vmiName,
+					metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				if vmi.Status.MigrationState == nil {
+					return false
+				}
+				if !vmi.Status.MigrationState.Completed {
+					return false
+				}
+				Expect(vmi.Status.MigrationState.Failed).To(BeFalse())
+
+				return true
+			}, 120*time.Second, time.Second).Should(BeTrue())
+		}
+		createDV := func() *cdiv1.DataVolume {
+			sc, exist := libstorage.GetRWOFileSystemStorageClass()
+			Expect(exist).To(BeTrue())
+			dv := libdv.NewDataVolume(
+				libdv.WithRegistryURLSource(cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskCirros)),
+				libdv.WithPVC(libdv.PVCWithStorageClass(sc),
+					libdv.PVCWithVolumeSize(size),
+				),
+			)
+			_, err := virtClient.CdiClient().CdiV1beta1().DataVolumes(ns).Create(context.Background(),
+				dv, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			return dv
+		}
+		createBlankDV := func() *cdiv1.DataVolume {
+			sc, exist := libstorage.GetRWOFileSystemStorageClass()
+			Expect(exist).To(BeTrue())
+			dv := libdv.NewDataVolume(
+				libdv.WithBlankImageSource(),
+				libdv.WithPVC(libdv.PVCWithStorageClass(sc),
+					libdv.PVCWithVolumeSize(size),
+				),
+			)
+			_, err := virtClient.CdiClient().CdiV1beta1().DataVolumes(ns).Create(context.Background(),
+				dv, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			return dv
+		}
+		createVM := func(dv *cdiv1.DataVolume) *virtv1.VirtualMachine {
+			vmi := libvmi.New(
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+				libvmi.WithResourceMemory("128Mi"),
+				libvmi.WithDataVolume("disk0", dv.Name),
+				libvmi.WithCloudInitNoCloudEncodedUserData(("#!/bin/bash\necho hello\n")),
+			)
+			vmi.Namespace = ns
```

Comment:
I think this can be dropped simply using `ns` during creation
```
vm, err := virtClient.VirtualMachine(ns).Create(context.Background(), vm, metav1.CreateOptions{})
```

Reply:
No, it cannot, otherwise I get this error during the tests:
```  
[FAILED] Unexpected error:
      <*errors.errorString | 0xc004accb80>: 
      an empty namespace may not be set when a resource name is provided
      {
          s: "an empty namespace may not be set when a resource name is provided",
      }
  occurred
  In [It] at: tests/libwait/wait.go:76 @ 05/30/24 07:41:47.243
```

Reply:
Then a simply nit:
`libvmi.WithNamespace(ns)`